### PR TITLE
[FR-180] NextJS Fix Channel Highlighter Position

### PR DESCRIPTION
**Pull request by tylergeorges**

### CODE REVIEW

**Review by mason-rogers — APPROVED** [review]

**Review by advaith1 — CHANGES_REQUESTED** [review]
pretty good, but glitches when collapsing categories:

https://github.com/widgetbot-io/embed/assets/11778454/7a095d95-350d-45dc-9558-a91d2425f5a9

File: src/components/Sidebar/ChannelsSidebar/Category/ChannelsContainer/Channel.tsx [hunk 1]
```diff
@@ -32,19 +32,39 @@ export const Channel = forwardRef<HTMLAnchorElement, ChannelNameProps>(
...
-      setCurrentChannelYPos(e.currentTarget.offsetTop);
-      setInitChannelYPos(e.currentTarget.offsetTop);
+      if (isActive) return;
+
+      const target = e.currentTarget;
+
+      if (channelClickTimeoutRef.current) clearTimeout(channelClickTimeoutRef.current);
+
+      channelClickTimeoutRef.current = setTimeout(() => {
+        //   // console.log(target.offsetTop);
```
Comment (advaith1) [inline src/components/Sidebar/ChannelsSidebar/Category/ChannelsContainer/Channel.tsx:63]: commented out code

File: src/components/Sidebar/ChannelsSidebar/Category/index.tsx [hunk 2]
```diff
@@ -47,20 +47,26 @@ export const Category = ({ category }: CategoryProps) => {
       setInitChannelYPos(currentChannelRef.current.offsetTop);
     }
 
-    if (channelsConRef.current) {
+    if (channelsConRef.current && !channelsConHeight) {
+      // if (channelsConRef.current && !channelsConHeight) {
```
Comment (advaith1) [inline src/components/Sidebar/ChannelsSidebar/Category/index.tsx:51]: commented out code

**Review by advaith1 — CHANGES_REQUESTED** [review]
still a bit misaligned for collapsed categories

https://github.com/widgetbot-io/embed/assets/11778454/fdca126d-eb77-4a2d-ab89-7299ec30e1f3

**Review by advaith1 — APPROVED** [review]

**Pull request merged**